core/watcher/watchertest/watcher.go

```diff
@@ -44,3 +47,49 @@ func DirtyKill[T any](c *gc.C, w watcher.Watcher[T]) {
 		c.Logf("ignoring failed to close for watcher")
 	}
 }
+
+// WatcherC embeds a gocheck.C and adds methods to help
+// verify the behaviour of generic watchers.
+type WatcherC[T any] struct {
+	C       *gc.C
+	Watcher watcher.Watcher[T]
+}
+
+// NewWatcherC() returns a WatcherC[T].
+func NewWatcherC[T any](c *gc.C, w watcher.Watcher[T]) WatcherC[T] {
+	return WatcherC[T]{
+		C:       c,
+		Watcher: w,
+	}
+}
+
+// AssertNoChange verifies that no changes are received from the watcher during
+// a testing.ShortWait time.
+func (w *WatcherC[T]) AssertNoChange() {
+	select {
+	case actual, ok := <-w.Watcher.Changes():
+		w.C.Fatalf("watcher sent unexpected change: (%v, %v)", actual, ok)
+	case <-time.After(testing.ShortWait):
+	}
+}
+
+// AssertChange asserts the given changes was reported by the watcher,
+// but does not assume there are no following changes.
```

Comment:
**todo:** This comment is out-dated. Assert change doesn't assert anything, since assertion as been moved to the function passed as parameter

**suggestion:** Give a proper name for type `func(c *gc.C, receiverd []T) bool`, like `WatcherAssert` and rename this one `Check`

aka:

```go
func (w *WatcherC[T]) Check(assertion WatcherAssert[T]) { /* ... */ }
```